### ISSUE TITLE
Provide an example of `publicDomain` usage

### ISSUE THREAD

**Issue by kilburn** [kilburn/gatsby-source-s3-asset#10]
The explanation of the `publicDomain` setting and its purpose doesn't seem complete enough.

Expand on that please.

**Comment by stcalica** [comment]
> Look at the publicDomain parameter. The way I'm using the plugin is with a private bucket that is also served through CloudFront. Hence the plugin fetches info from S3 (cloudfont doesn't have this functionality) but then links everything through the publicDomain which is a CloudFront distribution acting as CMS for the assets.

To continue the discussion, what your doing here is taking them from S3 and adding it to CloudFront (denoted by the public domain)?

**Comment by kilburn** [comment]
I am not "adding" them to CloudFront. I have a CloudFront distribution that acts as a CDN for the S3 bucket.

Gatsby lists the files from the bucket, and then generates URLs using the CloudFront-based URL, not the AWS URL. This also means I could apply restrictions at the CloudFront level should I want to (but I'm not because the files are indeed all public)

**Comment by stcalica** [comment]
Ah okay, you are saying that if there is no distribution then to make the asset URL from S3 rather than Cloudfront (or some other S3 API Compliant CDN)?

Sorry I am just learning about "S3 compliance" and that may be the source of my confusion here.

**Comment by kilburn** [comment]
Yes, if you don't specify any "public domain" the plugin will generate regular URLs pointing directly to S3. Something like:

https://your-bucket.s3-eu-west-1.amazonaws.com/your/file.ext

Whereas if you specify `publicDomain: somedomain.tld`, the plugin will generate URLs of the form:

https://somedomain.tld/your/file.ext

The plugin itself does nothing to ensure that the file exists on that public domain though, that is your job and can be achieved in many different ways (most notably by using CDN services).

**Comment by stcalica** [comment]
Ah, so some setup is required prior to this. 
So that is a way to hide the private S3 source and expose only the CDN. 
Awesome! 
I think this should be documented as well, to be honest.
